### PR TITLE
Mesh Export

### PR DESCRIPTION
**Pull request by bennyguo**
Support textured mesh export. Major changes:

- Add `--export` option in `launch.py`.
- Add a new class of model: exporters. `system.exporter_type` and `system.exporter` are specified when exporting.
- Add mesh exporter which now supports exporting obj files.
- Add a `BaseLift3DSystem` that contains common configurations like geometry, material, background, guidance, prompt_processor, and exporter.

### CODE REVIEW

File: threestudio/models/exporters/mesh_exporter.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+                params["map_Bump"] = uv_padding(geo_out["normal"])
+
+            if "albedo" in mat_out:
+                params["map_Kd"] = uv_padding(mat_out["albedo"])
+            else:
+                threestudio.warn(
+                    "save_texture is True but no albedo texture found, using default white texture"
+                )
+            # TODO: map_Ks
+        return [ExporterOutput(save_name="model.obj", save_type="obj", params=params)]
```
[the remark this answers is not in the document]
Comment (bennyguo) [reply threestudio/models/exporters/mesh_exporter.py:125]: done

File: threestudio/models/geometry/volume_grid.py [hunk 2]
```diff
@@ -155,3 +159,18 @@ def forward_level(
         self, field: Float[Tensor, "*N 1"], threshold: float
     ) -> Float[Tensor, "*N 1"]:
         return -(field - threshold)
+
+    def export(self, points: Float[Tensor, "*N Di"], **kwargs) -> Dict[str, Any]:
+        out: Dict[str, Any] = {}
+        if self.cfg.n_feature_dims == 0:
+            return out
+        points_unscaled = points
```
Comment (thuliu-yt16) [inline threestudio/models/geometry/volume_grid.py:167]: delete this line

Comment (thuliu-yt16) [inline threestudio/models/geometry/volume_grid.py:167]: delete this line

**Review by thuliu-yt16 — CHANGES_REQUESTED** [review]

File: threestudio/models/geometry/volume_grid.py (hunk 2 above)
Comment (bennyguo) [reply threestudio/models/geometry/volume_grid.py:167]: In `implicit-volume`, if `n_feature_dims==0` there will be no `feature_network`, so I return an empty dict. I think we better keep this to align with `implicit-volume`?

**Pull request merged**